dockstore-webservice/src/main/java/io/swagger/api/impl/ToolsImplCommon.java

```diff
@@ -321,21 +323,24 @@ public static void processImageDataForToolVersion(io.dockstore.webservice.core.T
         List<Checksum> trsChecksums = new ArrayList<>();
         if (version.getImages() != null && !version.getImages().isEmpty()) {
             version.getImages().forEach(image -> {
-                ImageData data = new ImageData();
-                image.getChecksums().forEach(checksum -> {
-                    Checksum trsChecksum = new Checksum();
-                    trsChecksum.setType(DOCKER_IMAGE_SHA_TYPE_FOR_TRS);
-                    trsChecksum.setChecksum(checksum.getChecksum());
-                    trsChecksums.add(trsChecksum);
-                });
-                //TODO: for now, all container images are Docker based
-                data.setImageType(ImageType.DOCKER);
-                data.setSize(image.getSize());
-                data.setUpdated(image.getImageUpdateDate());
-                data.setRegistryHost(image.getImageRegistry().getDockerPath());
-                data.setImageName(constructImageName(image));
-                data.setChecksum(trsChecksums);
-                toolVersion.getImages().add(data);
+                if (image.getImageRegistry() != null) {
+                    // avoid exception on null image registry
```

Comment:
I take it that a null image registry will be case for AppTools? If so, can you make the comment more specific, e.g., `AppTools don't have an image registry`? Otherwise it's not clear if it's a internal bug we're working around.

Reply:
This is an unrelated bug that was fixed before here https://github.com/dockstore/dockstore/pull/4346/files/5db53f46319fc93d7da3c76b6ae17a8f611611e2#diff-846e040caed0767d76be1edbffd8d114178f7e319e18f4f2730d3042253e96cdR292 but wasn't added to this section. We have some images in our db that don't have the image registry stored, although I don't know why.